OpenRA.Mods.Common/Traits/World/AutoSave.cs

```diff
@@ -0,0 +1,78 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[TraitLocation(SystemActors.World)]
+	[Desc("This trait allows setting a auto save options. Attach this to the World actor.")]
+	public class AutoSaveInfo : TraitInfo
+	{
+		public override object Create(ActorInitializer init) { return new AutoSave(init.Self, this); }
+	}
+
+	public class AutoSave : ITick
+	{
+		const string AutoSavePattern = "auto-save@";
+		const string SaveFileExtension = ".orasav";
+		readonly DirectoryInfo autoSaveDirectoryInfo;
+		readonly string saveFolderPath;
+		public uint TicksUntilAutoSave;
+		public AutoSave(Actor self, AutoSaveInfo info)
+		{
+			var autoSaveValue = Game.Settings.GamePlay.AutoSaveInterval;
+			var ticksAtWhichToAutosave = 1000 / self.World.Timestep * autoSaveValue;
+			TicksUntilAutoSave = (uint)ticksAtWhichToAutosave;
+
+			var mod = Game.ModData.Manifest;
+
+			saveFolderPath = Path.Combine(Platform.SupportDir, "Saves", mod.Id, mod.Metadata.Version);
+
+			autoSaveDirectoryInfo = new DirectoryInfo(saveFolderPath);
+		}
+
+		void ITick.Tick(Actor self)
+		{
+			var autoSaveValue = Game.Settings.GamePlay.AutoSaveInterval;
+			var autoSaveFileLimit = Game.Settings.GamePlay.AutoSaveNo;
+
+			if (self.World.Paused || autoSaveValue == 0 || autoSaveFileLimit == 0) return;
+
+			if (TicksUntilAutoSave != 0)
+			{
+				TicksUntilAutoSave--;
+				return;
+			}
+
+			if (GetAutoSaveFiles().Count() == autoSaveFileLimit)
+			{
+				GetAutoSaveFiles().OrderBy(f => f.CreationTime).FirstOrDefault()?.Delete();
+			}
+
+			var fileName = AutoSavePattern + DateTime.Now.ToString("dd_MM_yy-HH:mm:ss", CultureInfo.CurrentCulture) + SaveFileExtension;
```

Comment:
```suggestion
			var dateTime = DateTime.Now.ToString("yyyy-MM-ddTHHmmssZ", CultureInfo.InvariantCulture);
			var fileName = $"{AutoSavePattern}{dateTime}{SaveFileExtension}";
```

Unfortunately interpolated string handlers require C# 10.0, but that should be coming soon to OpenRA!

```
string.Create(CultureInfo.CurrentCulture, $"{AutoSavePattern}{DateTime.Now:dd_MM_yy-HH:mm:ss}{SaveFileExtension}");
```